### PR TITLE
Handle v2 version of parameters file during deployment

### PR DESCRIPTION
**Pull request by vivsriaus**
Handle v2 version of parameters file during deployment

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vivsriaus__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (visriniv). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: test/data/arm-deployment-parametersv2.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (amarzavery) [inline test/data/arm-deployment-parametersv2.json]: @vivsriaus - if my understanding is correct, then v2 should have top level key as "parameters" like this:
{
  "parameters" : 
  {
    "siteName": {
      "value": "xDeploymentTestSite16754"
    },
    "hostingPlanName": {
      "value": "xDeploymentTestHost24764"
    },
    "siteLocation": {
      "value": "West US"
    },
    "sku": {
      "value": "Free"
    },
    "workerSize": {
      "value": 0
    }
  }
}
```
That will test v2 in true sense.

File: test/commands/arm/group/arm.group.deployment-tests.js [hunk 2]
```diff
@@ -87,13 +87,16 @@ describe('arm', function () {
 
     function setUniqParameterNames(suite, filename) {
       //no need to create unique parameter values in playbackmode
-      var parameters = JSON.parse(fs.readFileSync(filename).toString());
+      var deploymentParameters = JSON.parse(fs.readFileSync(filename).toString());
+      if (deploymentParameters.parameters) {
```
Comment (amarzavery) [inline test/commands/arm/group/arm.group.deployment-tests.js]: Why would you add this in test code? The command code should handle it for you, right?

**Comment by markcowl** [comment]
@vivsriaus Please respond to feedback

**Comment by markcowl** [comment]
@vivsriaus I believe Amar is correct about V2 parameters files (they also have a schema and contenVersion property).  Please add in a proper test for v2 parameters.  You cna find a schema for V2 parameter files here:  https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/TemplateFileParameterv2.cs .  Please correct the issues or respond to review feedback.

File: test/data/arm-deployment-parametersv2.json (hunk 1 above)
Comment (vivsriaus) [inline test/data/arm-deployment-parametersv2.json]: @amarzavery My bad, you're correct. It should include the parameters element. I've fixed this now with my latest commit.

File: test/commands/arm/group/arm.group.deployment-tests.js (hunk 2 above)
Comment (vivsriaus) [inline test/commands/arm/group/arm.group.deployment-tests.js]: setUniqParameterNames is called before we execute the command in our tests, to set unique names for parameters. So, this is required to differentiate v2 and v1 parameters file.

**Comment by ahmetalpbalkan** [comment]
@amarzavery I still fail to deploy templates whose parameters file starts like the following:

```
{
  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
```

and getting error:

```
+ Initializing template configurations and parameters
+ Creating a deployment
error:   The request content was invalid and could not be deserialized: 'Error converting value "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#" to type 'Microsoft.WindowsAzure.ResourceStack.Frontdoor.Data.Definitions.DeploymentParameterDefinition'. Path 'properties.parameters.$schema', line 1, position 15622.'.
info:    Error information has been recorded to /Users/alp/.azure/azure.err
error:   group deployment create command failed
```

**Pull request merged**